### ISSUE TITLE
Revert certain reverts

### ISSUE THREAD

**Issue by emanuil-tolev** [DOAJ/doaj#142]
A couple of commits were merged into master by mistake so when rolling phase2b out, revert:

Journal ToC-s:
- c7d02f6233de0dcdce8d89565b6257a23c4b9e8f
- d478ee22074a9ac58a31146e6e786ed2828b486d

---

I wrote the impersonation thing in 15 minutes before @richard-jones had a chance to comment, but I agree with the comments that it should be written and rolled out as part of a bigger overhaul.

If the code from is to be used in the overhaul of the user management, then revert this commit before starting development on it: b5176a22a426e27d4ff221ddd8cf0b8710e7e9c3

If the revert-revert to phase2b recommended above is done then when merging phase2b into master for a rollout, revert 5b29bc421a3ba08632da77b72ef0f10a0bf42239 .
